Orderbook v2

nakul1010 opened the pull request.

Gauthamastro committed:
Initial Commit

Gauthamastro committed:
Boiler plate

Gauthamastro committed:
Boiler plate done.

Gauthamastro committed:
Gossip

Gauthamastro committed:
Somewhere in the unknown territory.

Gauthamastro committed:
Snapshot storage and retriveal looks like it will work.

Gauthamastro committed:
Logic for syncing is almost done

Gauthamastro committed:
Hooked up network notification events

Gauthamastro committed:
Hooked up network notification events

Gauthamastro committed:
We are somewhere

Gauthamastro committed:
cargo fix

Gauthamastro committed:
Added withdrawal handling

Gauthamastro committed:
BLS keys

Gauthamastro committed:
BLS keys

Gauthamastro committed:
Future send issue is fixed now

zktony committed:
Updated snasphot ext

Gauthamastro committed:
Almost there

Gauthamastro committed:
bls is compiling

Gauthamastro committed:
Merge branch 'gj-ob' into zk-impl-snapshot

Gauthamastro committed:
BLS host functions hooked up

Gauthamastro committed:
Re-impl Snapshot Ext (#659)

## Describe your changes
* Re-implemented Snapshot Ext.
* Impl SessionHnadler trait.
* Removed Enclave code.

Gauthamastro committed:
Lets go.

Gauthamastro committed:
Submit snapshot here we go.

Gauthamastro committed:
Almost there

Gauthamastro committed:
Almost there

Gauthamastro committed:
All set

Gauthamastro committed:
Remove old gossipmessage

Gauthamastro committed:
Cargo fix

Gauthamastro committed:
Something broke

Gauthamastro committed:
Let's test it.

Gauthamastro committed:
lets test it for sure now

Gauthamastro committed:
Handle vector of trades

Gauthamastro committed:
Hook up the metrics

Gauthamastro committed:
local test network code

Gauthamastro committed:
Add Eq for IngressMessages

Gauthamastro committed:
Add Eq for IngressMessages

Gauthamastro committed:
Bls bug fixes

Gauthamastro committed:
Bls bug fixes

Gauthamastro committed:
Bls bug fixes

felixfaisal committed:
fix: update chain_spec

felixfaisal committed:
fix: add orderbook session keys

felixfaisal committed:
fix: add scripts for starting chain

Gauthamastro committed:
feat: scripts for starting chain  (#665)

This PR contains scripts for starting the chain with orderbook keys,
with Validators and 1 Full Node

Gauthamastro committed:
Handle bad txns

Gauthamastro committed:
Merge remote-tracking branch 'origin/gj-ob' into gj-ob

# Conflicts:
#	node/src/chain_spec.rs

Gauthamastro committed:
Remove test_bls

Gauthamastro committed:
Added withdrawal verify

Gauthamastro committed:
Moving stuff

Gauthamastro committed:
Moving stuff

Gauthamastro committed:
Moving stuff

Gauthamastro committed:
Moving stuff

Gauthamastro committed:
Moving stuff

Gauthamastro committed:
Moving stuff

Gauthamastro committed:
Moving stuff

Gauthamastro committed:
Moving stuff

Gauthamastro committed:
Moving stuff

Gauthamastro committed:
Moving stuff

Gauthamastro committed:
Moving stuff

Gauthamastro committed:
Metrics

nuel77 committed:
fix ord for order

Gauthamastro committed:
test macros

Gauthamastro committed:
Merge remote-tracking branch 'origin/gj-ob' into gj-ob

Gauthamastro committed:
Testing framework is setup

Gauthamastro committed:
Testing...

Gauthamastro committed:
Almost done.

Gauthamastro committed:
Add max withdrawals per snapshot

Gauthamastro committed:
State sync via fullnode gossip

Gauthamastro committed:
orderbook client compiles

Gauthamastro committed:
everything compiles

Gauthamastro committed:
bls wild exploration

Gauthamastro committed:
Fixed bls keys

nuel77 committed:
increase round_off precision

nuel77 committed:
Merge remote-tracking branch 'origin/gj-ob' into gj-ob

Gauthamastro committed:
load testing files

Gauthamastro committed:
Merge remote-tracking branch 'origin/gj-ob' into gj-ob

Gauthamastro committed:
BLS sign works now

Gauthamastro committed:
Add min_volume to TradingPairConfig

Gauthamastro committed:
ALmost there

Gauthamastro committed:
ALmost there

nuel77 committed:
debug

Gauthamastro committed:
ALmost there

nuel77 committed:
debug

nuel77 committed:
Merge remote-tracking branch 'origin/gj-ob' into gj-ob

Gauthamastro committed:
logic for handling validator session change.

Gauthamastro committed:
genesis migration logic

nakul1010 committed:
added Test cases

zktony committed:
Added tests

nakul1010 committed:
Merge branch 'ob-client-tests' of https://github.com/Polkadex-Substrate/Polkadex into nap-v2-test-ob

# Conflicts:
#	clients/orderbook/src/lib.rs

felixfaisal committed:
fix: created keystore struct

felixfaisal committed:
fix: update generate_pair_and_store

felixfaisal committed:
fix: update sign function to derive child from master

felixfaisal committed:
fix: update test to include junction

Gauthamastro committed:
fix: bls verify bug (#673)

felixfaisal committed:
fix: build fix, using std feature for segregation

Gauthamastro committed:
fix: build fix, using std feature for segregation (#674)

nakul1010 committed:
added Test cases

zktony committed:
Added test cases

nakul1010 committed:
Merge branch 'ob-client-tests' of https://github.com/Polkadex-Substrate/Polkadex into nap-v2-test-ob

# Conflicts:
#	clients/orderbook/src/tests.rs

nakul1010 committed:
added ZK TC

felixfaisal committed:
fix: remove shutdown ingress message

felixfaisal committed:
refactor: apply rustfmt

nakul1010 committed:
boiler plate

nakul1010 committed:
serialize before sending data back

nakul1010 committed:
snapshot generation logic added

nakul1010 committed:
fix ci

nakul1010 committed:
commented out ci builds and code, created issues with todos for commented out code

nakul1010 committed:
Merge branch 'gj-ob' of https://github.com/Polkadex-Substrate/Polkadex into nap-fix-730

# Conflicts:
#	.github/workflows/ci.yml
#	pallets/liquidity/src/lib.rs
#	pallets/ocex/src/lib.rs

nakul1010 committed:
updated comment

nakul1010 committed:
updated comment

nakul1010 committed:
The "gj-ob" branch has been added to trigger continuous integration (CI) whenever a pull request (PR) is raised.

nakul1010 committed:
Merge branch 'gj-ob' of https://github.com/Polkadex-Substrate/Polkadex into nap-fix-730

nakul1010 committed:
Merge branch 'gj-ob' of https://github.com/Polkadex-Substrate/Polkadex into nap-v2-test-ob

# Conflicts:
#	clients/orderbook/src/lib.rs
#	pallets/ocex/src/lib.rs

serhii-temchenko committed:
Temporary fix of CI workflow

nakul1010 committed:
added docs for test cases

nakul1010 committed:
Merge branch 'gj-ob' of https://github.com/Polkadex-Substrate/Polkadex into nap-fix-730

nakul1010 committed:
comment out clippy

nakul1010 committed:
fix ci file

nakul1010 committed:
added formatting check back

nakul1010 committed:
commented out clippy check from CI

nakul1010 committed:
Merge branch 'gj-ob' of https://github.com/Polkadex-Substrate/Polkadex into nap-fix-730

# Conflicts:
#	.github/workflows/ci.yml

nakul1010 committed:
Merge branch 'gj-ob' of https://github.com/Polkadex-Substrate/Polkadex into nap-v2-test-ob

nakul1010 committed:
fix TC

nakul1010 committed:
Merge branch 'gj-ob' of https://github.com/Polkadex-Substrate/Polkadex into nap-fix-730

nakul1010 committed:
Merge branch 'gj-ob' of https://github.com/Polkadex-Substrate/Polkadex into nap-v2-test-ob

nakul1010 committed:
Merge branch 'gj-ob' of https://github.com/Polkadex-Substrate/Polkadex into nap-v2-recovery

nakul1010 committed:
Created RW lock for existing fields & updated OrderbookRPC

zktony committed:
Added code to process withdraw

zktony committed:
Fixed formatting

nakul1010 committed:
removed unused parameters

nakul1010 committed:
get_allowlisted_assets OB api created

nakul1010 committed:
completed RPC

nakul1010 committed:
snapshot generation logic (#681)

Fixes
[#730](https://github.com/orgs/Polkadex-Substrate/projects/22/views/1?filterQuery=nakul1010&pane=issue&itemId=23972760)

nakul1010 committed:
Merge branch 'gj-ob' of https://github.com/Polkadex-Substrate/Polkadex into nap-v2-recovery

# Conflicts:
#	pallets/ocex/src/lib.rs
#	primitives/orderbook/src/lib.rs
#	runtime/src/lib.rs

serhii-temchenko committed:
Code style fixes

serhii-temchenko committed:
Renamed tests module and updated tests names

nakul1010 committed:
completed RPC

nakul1010 committed:
Merge branch 'gj-ob' of https://github.com/Polkadex-Substrate/Polkadex into nap-v2-test-ob

# Conflicts:
#	clients/orderbook/src/lib.rs

nakul1010 committed:
handled conflicts fix TC

serhii-temchenko committed:
Removed unused imports in tests

serhii-temchenko committed:
Scoped tests only related method to be accessible only for tests

serhii-temchenko committed:
Code style fixes

nakul1010 committed:
Test cases for V2 (#675)

Partially resolves
[703](https://github.com/Polkadex-Substrate/orderbook/issues/704)

- [x]  `register_main`
    - Access to DB
- [x]  `add_proxy`
  - Access to DB
- [x]  `remove_proxy`
  - Access to DB
- [x]  `deposit`
  - Access to DB
- [x]  `process_trade`
  - Access to DB
- [x] `process_chunk`
- [x] `store_snapshot`
- [x] `load_snapshot`
- [x] `add_balance`
- [x] `sub_balance`

nakul1010 committed:
fixed worker

nakul1010 committed:
Merge branch 'gj-ob' of https://github.com/Polkadex-Substrate/Polkadex into nap-v2-recovery

nakul1010 committed:
handled conflicts, test_single_worker test case failing

nakul1010 committed:
fix trie

nakul1010 committed:
made last_block_snapshot_generated a read_write

nakul1010 committed:
refactor lock

nakul1010 committed:
added inline comments

nakul1010 committed:
fmt added

serhii-temchenko committed:
Code style fixes and cleanups

serhii-temchenko committed:
Corrected properties names

serhii-temchenko committed:
Added and used db alias type

serhii-temchenko committed:
Merge branch 'gj-ob' into fx-remove-shutdown

serhii-temchenko committed:
Added missing required impoprt

serhii-temchenko committed:
Cleaned up benchmarks for removed shutdown extrinsic

serhii-temchenko committed:
Cleaned up shutdown extrinsic tests

nakul1010 committed:
fix comment

nakul1010 committed:
fix reviews

Gauthamastro committed:
V2 recovery (#688)

Fixes Part 1 of
[#733](https://github.com/Polkadex-Substrate/orderbook/issues/733)

zktony committed:
Updated pallet

nakul1010 committed:
Process withdrawals from Snapshot (#687)

## Describe your changes
Process withdrawals from Snapshot.

## Issue ticket number and link
Will close https://github.com/Polkadex-Substrate/orderbook/issues/755

nakul1010 committed:
ordering changed

nuel77 committed:
Fix ordering (#691)

Fixes [771](https://github.com/Polkadex-Substrate/orderbook/issues/771)

serhii-temchenko committed:
Merge branch 'gj-ob' into fx-remove-shutdown

# Conflicts:
#	pallets/ocex/src/lib.rs

nakul1010 committed:
Merge branch 'Develop' of https://github.com/Polkadex-Substrate/Polkadex into gj-ob

# Conflicts:
#	.github/workflows/ci.yml
#	Cargo.lock
#	Cargo.toml
#	node/Cargo.toml

nakul1010 committed:
handled conflicts

serhii-temchenko commented on .github/workflows/ci.yml:
```diff
@@ -3,7 +3,7 @@ name: Continuous Integration
   pull_request: []
   push:
-    branches: [Develop, mainnet-release]
+    branches: [Develop, main-net-runtime, redux-ci-testing, gj-ob]
```
`redux-ci-testing` and `gj-ob` should be removed

serhii-temchenko replied:
Fixed

serhii-temchenko commented on client/Cargo.toml:
```diff
@@ -10,3 +10,4 @@ sc-executor = { workspace = true }
 node-polkadex-runtime = { path = "../runtime", version = "4.0.0" }
 frame-benchmarking = { workspace = true }
 thea-primitives = {path = "../thea-primitives"}
+bls-primitives = {path = "../primitives/bls-primitives"}
```
Formatting (spaces before `}` and after `{`)

serhii-temchenko replied:
Fixed

serhii-temchenko commented on Cargo.toml:
```diff
@@ -135,7 +144,7 @@ substrate-wasm-builder = { git = "https://github.com/paritytech/substrate", bran
 pallet-nomination-pools = { git = "https://github.com/paritytech/substrate", branch = "polkadot-v0.9.28", default-features = false }
 grandpa-primitives = {  git = "https://github.com/paritytech/substrate", branch = "polkadot-v0.9.28", default-features = false, package="sp-finality-grandpa" }
-polkadex-primitives = { tag = "polkadot-v0.9.28v0.1.14", git = "https://github.com/Polkadex-Substrate/polkadex-primitives.git", default-features = false }
+polkadex-primitives = { path="primitives/polkadex", default-features = false }
```
Formatting. Spaces.

serhii-temchenko replied:
Fixed

serhii-temchenko commented on misc/load-testing/Cargo.toml:
```diff
@@ -0,0 +1,11 @@
+[dependencies]
+tokio = {version="1.26.0", features = ["rt-multi-thread"]}
+jsonrpsee = {version="0.16.2", features = ["ws-client"]}
+orderbook-primitives = {path ="../../primitives/orderbook"}
```
Formatting + empty line at the end of file

serhii-temchenko replied:
Fixed

serhii-temchenko commented on node/src/main.rs:
```diff
@@ -1,4 +1,4 @@
-#![deny(unused_crate_dependencies)]
+// #![deny(unused_crate_dependencies)]
```
Fix deps and uncomment it back.

serhii-temchenko commented on pallets/liquidity/src/lib.rs:
```diff
@@ -20,11 +20,13 @@
-mod tests;
 
-#[cfg(test)]
-mod mock;
+// ToDo: Issue 682
+// #[cfg(test)]
+// mod tests;
+//
+// #[cfg(test)]
+// mod mock;
```
Should be addressed asap (after merge?)

Gauthamastro replied:
This one after merge. It came from Develop and is not part of this PR.

serhii-temchenko commented on pallets/ocex/Cargo.toml:
```diff
@@ -20,13 +20,15 @@ frame-system = { default-features = false, workspace = true }
+sp-core = { workspace = true, default-features = false }
 liquidity = { path = "../liquidity", default-features = false }
+orderbook-primitives = {path="../../primitives/orderbook", default-features = false}
+bls-primitives = {path="../../primitives/bls-primitives", default-features = false}
```
Formatting

serhii-temchenko replied:
Fixed

serhii-temchenko commented on pallets/ocex/src/lib.rs:
```diff
@@ -20,29 +20,34 @@
+// use sp_runtime::traits::One;
+//
+// #[cfg(feature = "runtime-benchmarks")]
+// mod benchmarking;
```
Should be addressed asap after merging this PR?

serhii-temchenko replied:
Fixed in this pr

serhii-temchenko commented on pallets/ocex/src/lib.rs:
```diff
@@ -133,13 +139,73 @@ pub mod pallet {
+		type Call = Call<T>;
+
+		fn validate_unsigned(source: TransactionSource, call: &Self::Call) -> TransactionValidity {
+			sp_runtime::print("Entering validate unsigned....");
```
Do we need this print?

Gauthamastro replied:
Yes, until we clean up before creating a release to the main. We have to see this in action in testnet.

serhii-temchenko commented on pallets/ocex/src/lib.rs:
```diff
@@ -133,13 +139,73 @@ pub mod pallet {
+						// 	&signature,
+						// ) {
+						// 	return InvalidTransaction::Custom(13).into()
+						// }
```
Whats about this one?

serhii-temchenko replied:
Resolved in this PR

serhii-temchenko commented on pallets/ocex/src/lib.rs:
```diff
@@ -133,13 +139,73 @@ pub mod pallet {
+						// }
+					},
+				}
+				sp_runtime::print("Signature successfull");
```
Do we need this? Please remove.

serhii-temchenko replied:
Moved to the line number 1450?

serhii-temchenko replied:
Same as [here](https://github.com/Polkadex-Substrate/Polkadex/pull/693#discussion_r1159249073).

serhii-temchenko commented on pallets/ocex/src/lib.rs:
```diff
@@ -701,84 +756,43 @@ pub mod pallet {
 		}
 
-		/// Extrinsic used by enclave to submit balance snapshot and withdrawal requests
+		// TODO: Benchmark change_pending_withdrawal_limit (Issue number 683)
```
Still pending

serhii-temchenko replied:
Fixed

serhii-temchenko commented on pallets/ocex/src/lib.rs:
```diff
@@ -701,84 +756,43 @@ pub mod pallet {
+		/// wants to set.
 		#[pallet::weight(<T as Config>::WeightInfo::submit_snapshot())]
-		pub fn submit_snapshot(
+		pub fn change_pending_withdrawal_limit(
```
Used wrong weight info

serhii-temchenko replied:
Fixed

serhii-temchenko commented on pallets/ocex/src/lib.rs:
```diff
@@ -701,84 +756,43 @@ pub mod pallet {
+		/// * `new_snapshot_interval_block`: The new block interval at which snapshot should  be
+		/// generated.
+		#[pallet::weight(<T as Config>::WeightInfo::submit_snapshot())]
+		pub fn change_snapshot_interval_block(
```
Used wrong weight

serhii-temchenko replied:
Fixed

serhii-temchenko commented on pallets/ocex/src/lib.rs:
```diff
@@ -701,84 +756,43 @@ pub mod pallet {
-		#[doc(hidden)]
-		#[pallet::weight(<T as Config>::WeightInfo::insert_enclave(1))]
-		pub fn insert_enclave(origin: OriginFor<T>, enclave: T::AccountId) -> DispatchResult {
+		// TODO: Benchmark change_snapshot_interval_block (Issue number 683)
```
Still pending

serhii-temchenko replied:
Fixed

serhii-temchenko commented on pallets/ocex/src/lib.rs:
```diff
@@ -1040,28 +1022,82 @@ pub mod pallet {
-			T::GovernanceOrigin::ensure_origin(origin)?;
-			<CertificateValidity<T>>::put(certificate_valid_until);
+		/// Submit Snapshot Summary
+		#[pallet::weight(10000)]
```
No benchmarks
No documentation

serhii-temchenko replied:
Benchmark part fixed. Documentation still pending.

serhii-temchenko commented on .github/workflows/ci.yml:
```diff
@@ -75,14 +75,19 @@ jobs:
-        run: cargo clippy -- -D warnings
+      #       ToDo: After fixing issue #684 uncomment clippy
+      #      - name: Check Clippy
+      #        run: cargo clippy -- -D warnings
```
Still not resolved. Should be addressed asap (after merge of this PR?)

serhii-temchenko replied:
Build part with formatting fixed and corrected. Clippy still remaining.

Gauthamastro replied:
New issue

serhii-temchenko commented on .gitignore:
```diff
@@ -17,3 +18,4 @@
 
 .idea
 
+/polkadex/
```
Why is it here?

Gauthamastro replied:
Bls primitives unit tests create a folder named polkadex.

serhii-temchenko commented on clients/orderbook/src/tests.rs:
```diff
@@ -0,0 +1,710 @@
+	workers.for_each(|_| async move {})
+}
+
+// TODO: Make this work
+// use sc_network_gossip::Network as GossipNetwork;
+// pub fn setup_one<B, BE, C, SO, N, R>(api: Arc<R>, is_validator: bool) -> (ObWorker<B, BE, C, SO,
+// N, R>, UnboundedSender<ObMessage>) where
+//     B: BlockT,
+//     BE: Backend<B>,
+//     C: Client<B, BE>,
+//     R: ProvideRuntimeApi<B>,
+//     R::Api: ObApi<B>,
+//     N: GossipNetwork<B> + Clone + Send + Sync + 'static + SyncOracle,
+//
+// {
+//     let testnet = ObTestnet::new(1, 0);
+//     let peer = &testnet.peers[0];
+//     let (rpc_sender, rpc_receiver) = futures::channel::mpsc::unbounded();
+//
+//     let worker_params = WorkerParams {
+//         client: peer.client().as_client(),
+//         backend: peer.client().as_backend(),
+//         runtime: api,
+//         sync_oracle: peer.network_service().clone(),
+//         network: peer.network_service().clone(),
+//         protocol_name: Cow::from("blah"),
+//         is_validator,
+//         message_sender_link: rpc_receiver,
+//         metrics: None,
+//         _marker: Default::default(),
+//     };
+//
+//     let worker = ObWorker::new(worker_params);
+//
+//     (worker, rpc_sender)
+// }
```
Still not resolved

Gauthamastro replied:
This is not a priority, and it's just there to have a reference for future testing of clients if any. I will remove the TODO.

serhii-temchenko commented on clients/orderbook/src/tests.rs:
```diff
@@ -0,0 +1,710 @@
+}
+
+#[test]
+pub fn test_trie_insertion() {
+	let mut working_state_root = [0u8; 32];
+	let mut memory_db: MemoryDB<RefHasher, HashKey<RefHasher>, Vec<u8>> = Default::default();
+	{
+		let mut trie: TrieDBMut<ExtensionLayout> =
+			TrieDBMutBuilder::new(&mut memory_db, &mut working_state_root).build();
+
+		//trie.insert(b"ab".as_ref(),b"cd".as_ref()).unwrap();
+		trie.commit();
+	}
+
+	// {
+	//     let mut trie: TrieDBMut<ExtensionLayout> =
+	//         TrieDBMutBuilder::from_existing(&mut memory_db, &mut working_state_root)
+	//             .build();
+	//    assert_eq!(trie.get(b"ab".as_ref()).unwrap(), Some(b"cd".to_vec()))
+	// }
+
+	assert_ne!(working_state_root, [0u8; 32]);
```
Should be properly tested with expected real value and not just that it's different from the initial...

Plus all commented out code should be removed

serhii-temchenko commented on clients/orderbook/src/worker.rs:
```diff
@@ -0,0 +1,1289 @@
+	pub backend: Arc<BE>,
+	pub runtime: Arc<R>,
+	pub sync_oracle: SO,
+	// pub key_store: BeefyKeystore,
+	// pub links: BeefyVoterLinks<B>,
```
Commented out code should be removed.

serhii-temchenko commented on clients/orderbook/src/worker.rs:
```diff
@@ -0,0 +1,1289 @@
+	sync_oracle: SO,
+	is_validator: bool,
+	network: Arc<N>,
+	// key_store: BeefyKeystore,
```
Commented out code should be removed.

serhii-temchenko commented on clients/orderbook/src/worker.rs:
```diff
@@ -0,0 +1,1289 @@
+	pub working_state_root: Arc<RwLock<[u8; 32]>>,
+	// Known state ids
+	known_messages: BTreeMap<u64, ObMessage>,
+	// Links between the block importer, the background voter and the RPC layer.
+	// links: BeefyVoterLinks<B>,
```
Commented out code should be removed.

serhii-temchenko commented on clients/orderbook/src/worker.rs:
```diff
@@ -0,0 +1,1289 @@
+			client,
+			backend,
+			runtime,
+			// key_store,
```
Commented out code should be removed.

serhii-temchenko commented on clients/orderbook/src/worker.rs:
```diff
@@ -0,0 +1,1289 @@
+			runtime,
+			// key_store,
+			sync_oracle,
+			// links,
```
Commented out code should be removed.

serhii-temchenko commented on clients/orderbook/src/worker.rs:
```diff
@@ -0,0 +1,1289 @@
+			backend,
+			runtime,
+			sync_oracle,
+			// key_store,
```
Commented out code should be removed.

serhii-temchenko commented on clients/orderbook/src/worker.rs:
```diff
@@ -0,0 +1,1289 @@
+			gossip_engine,
+			gossip_validator,
+			memory_db,
+			// links,
```
Commented out code should be removed.

serhii-temchenko commented on clients/orderbook/src/worker.rs:
```diff
@@ -0,0 +1,1289 @@
+		summary.aggregate_signature = Some(signature.clone());
+		let bit_index = active_set.iter().position(|v| v == &signing_key.into()).unwrap();
+		set_bit_field(&mut summary.bitflags, bit_index as u16);
+		// println!("Signature: {:?}",summary.aggregate_signature.unwrap().0);
+		// println!("Signing key: {:?}",signing_key.0);
```
Commented out code should be removed.

serhii-temchenko replied:
Removed

serhii-temchenko commented on clients/orderbook/src/worker.rs:
```diff
@@ -0,0 +1,1289 @@
+					.cloned()
+					.collect::<Vec<PeerId>>();
+				peers.append(&mut fullnodes);
+				// TODO: Should we even send it out to everyone we know?
```
Still pending

Gauthamastro replied:
It's not pending, these are something to test in testnet and then tweak.

serhii-temchenko commented on clients/orderbook/src/worker.rs:
```diff
@@ -0,0 +1,1289 @@
+
+	pub async fn got_stids_via_gossip(&mut self, messages: &Vec<ObMessage>) -> Result<(), Error> {
+		for message in messages {
+			// TODO: DO signature checks here and handle reputation
```
Still pending

Gauthamastro replied:
Not required in this version, let that TODO be there.

serhii-temchenko commented on clients/orderbook/src/worker.rs:
```diff
@@ -0,0 +1,1289 @@
+	// Expects the set bits in the bitmap to be missing chunks
+	pub async fn want(&mut self, snapshot_id: &u64, bitmap: &Vec<u128>, remote: Option<PeerId>) {
+		// Only respond if we are a fullnode
+		// TODO: Should we respond if we are also syncing???
```
Still pending

Gauthamastro replied:
It's not pending, and these are something to test in testnet and then tweak.

serhii-temchenko commented on clients/orderbook/src/worker.rs:
```diff
@@ -0,0 +1,1289 @@
+						}
+					}
+				} else {
+					// TODO: Reduce reputation if else block is happens
```
Still pending

Gauthamastro replied:
Reputation is not part of this version but these are places we need to handle it in next version.

serhii-temchenko commented on clients/orderbook/src/worker.rs:
```diff
@@ -0,0 +1,1289 @@
+					self.runtime.runtime_api().get_snapshot_by_id(&at, *snapshot_id)
+				{
+					let chunk_indexes: Vec<u16> = return_set_bits(bitmap);
+					// TODO: Santiy check to ensure that indexing in to state_chunk_hashes doesn't
+					// cause a panic
```
Still pending

Gauthamastro replied:
fixed

serhii-temchenko commented on clients/orderbook/src/worker.rs:
```diff
@@ -0,0 +1,1289 @@
+						.and_modify(|status| {
+							*status = StateSyncStatus::Available;
+						})
+						.or_insert(StateSyncStatus::Available); //TODO: @Gatham please corss check this
```
Still pending

serhii-temchenko commented on clients/orderbook/src/worker.rs:
```diff
@@ -0,0 +1,1289 @@
+							},
+							Some(data) => {
+								match SnapshotSummary::decode(&mut &data[..]) {
+									// TODO: This is repeated block of code from snapshot() fn,
+									// make it a single function
```
Still pending

Gauthamastro replied:
Removed.

serhii-temchenko commented on pallets/ocex/src/lib.rs:
```diff
@@ -701,84 +756,43 @@ pub mod pallet {
 
 		//TODO: Benchmark set_snapshot
 		#[pallet::weight(<T as Config>::WeightInfo::submit_snapshot())]
-		pub fn set_snapshot(origin: OriginFor<T>, new_snapshot_id: u32) -> DispatchResult {
+		pub fn set_snapshot(origin: OriginFor<T>, new_snapshot_id: u64) -> DispatchResult {
```
Still pending. Wrong weight used?

serhii-temchenko replied:
Fixed

serhii-temchenko commented on pallets/ocex/src/lib.rs:
```diff
@@ -1040,28 +1022,82 @@ pub mod pallet {
+			if working_summary.signed_auth_indexes().len() >=
+				total_validators.saturating_mul(2).saturating_div(3)
+			{
+				// TODO: Do we need to verify aggregrate signature once more???
```
Still pending

Gauthamastro replied:
Fixed.

serhii-temchenko commented on primitives/bls-primitives/Cargo.toml:
```diff
@@ -0,0 +1,37 @@
+sp-application-crypto = { git = "https://github.com/paritytech/substrate.git", branch = "polkadot-v0.9.28", default-features = false }
+sp-core = { git = "https://github.com/paritytech/substrate.git", branch = "polkadot-v0.9.28", default-features = false }
+sp-std = { git = "https://github.com/paritytech/substrate.git", branch = "polkadot-v0.9.28", default-features = false }
+serde_json = {version = "1.0.94", optional = true}
+blst = {version = "0.3.10", default-features = false, optional=true}
```
Formatting

serhii-temchenko replied:
Fixed

serhii-temchenko commented on primitives/bls-primitives/Cargo.toml:
```diff
@@ -0,0 +1,37 @@
+blst = {version = "0.3.10", default-features = false, optional=true}
+parity-scale-codec = { workspace = true, default-features = false, features = ["max-encoded-len"] }
+scale-info = { workspace = true, default-features = false, features = ["derive"] }
+hex = {version="0.4.3", optional= true}
```
Formatting

serhii-temchenko replied:
Fixed

serhii-temchenko commented on primitives/bls-primitives/Cargo.toml:
```diff
@@ -0,0 +1,37 @@
+    "substrate-bip39",
+    "tiny-bip39",
+    "blst",
+    ]
```
No empty line

serhii-temchenko replied:
Fixed

serhii-temchenko commented on primitives/bls-primitives/src/application_crypto.rs:
```diff
@@ -0,0 +1,66 @@
+#[cfg(test)]
+mod tests {
+	use crate::crypto::sign;
+	use sp_application_crypto::RuntimePublic;
```
Unused

serhii-temchenko replied:
Fixed

serhii-temchenko commented on primitives/bls-primitives/src/crypto.rs:
```diff
@@ -0,0 +1,243 @@
+#[cfg(feature = "std")]
+use crate::{Error, Pair as BLSPair};
+use crate::{KeyStore, Public, Seed, Signature, BLS_DEV_PHRASE, DEV_PHRASE, DST};
```
BLS_DEV_PHRASE, DEV_PHRASE unused

serhii-temchenko replied:
Fixed

serhii-temchenko commented on primitives/bls-primitives/src/crypto.rs:
```diff
@@ -0,0 +1,243 @@
+
+	fn generate_pair(phrase: Option<Vec<u8>>) -> Public {
+		// generate a pair
+		let (pair, _seed, derive_junctions) = generate_pair_(phrase);
```
`derive_junctions` unused

serhii-temchenko replied:
Fixed

serhii-temchenko commented on primitives/bls-primitives/src/lib.rs:
```diff
@@ -0,0 +1,330 @@
+			true,
+			message.as_ref(),
+			DST.as_ref(),
+			&[], // TODO: wtf is this?
```
Still pending

Gauthamastro replied:
We need to handle it outside of this PR.

serhii-temchenko commented on primitives/orderbook/Cargo.toml:
```diff
@@ -0,0 +1,49 @@
+# See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html
+
+[dependencies]
+primitive-types = {version = "0.12.1", default-features =false, features = ["scale-info"]}
+parity-scale-codec = { workspace = true }
+scale-info = { workspace = true }
+sp-api = { workspace = true }
+sp-io = { workspace = true }
+sp-application-crypto = { workspace = true }
+sp-core = { workspace = true }
+sp-runtime = { workspace = true }
+sp-std = { workspace = true}
+polkadex-primitives = { workspace = true }
+chrono = {version = "0.4.23", optional = true}
+rand = {version = "0.8.5", optional = true}
+serde = { version = "1.0.152", optional = true, default-features = false}
+anyhow = { version = "1.0.69", default-features = false }
+rust_decimal = { git = "https://github.com/Polkadex-Substrate/rust-decimal.git", branch = "master", features = ["scale-codec"], default-features = false }
+bls-primitives = {path="../bls-primitives", default-features = false}
+libp2p-core = {version ="0.34.0", optional = true}
```
Formatting

serhii-temchenko replied:
Fixed

serhii-temchenko commented on primitives/orderbook/src/lib.rs:
```diff
@@ -0,0 +1,220 @@
+		&self.validators
+	}
+
+	// /// Return the validator set id.
+	// pub fn id(&self) -> ValidatorSetId {
+	//     self.id
+	// }
```
Please remove

serhii-temchenko commented on primitives/orderbook/src/types.rs:
```diff
@@ -0,0 +1,595 @@
+
+impl OrderState {
+	pub fn from(order: &Order) -> Self {
+		// TODO: compute correct filled qty
```
Still pending

serhii-temchenko commented on primitives/orderbook/src/types.rs:
```diff
@@ -0,0 +1,595 @@
+	// verify if we can update the order state, with the new state of order.
+	pub fn update(&mut self, order: &Order, price: Decimal, amount: Decimal) -> bool {
+		// Verify signature also here.
+		// TODO: FIX this.
```
Still pending

Gauthamastro replied:
Fixed

serhii-temchenko commented on primitives/orderbook/src/types.rs:
```diff
@@ -0,0 +1,595 @@
+
+	// Verifies the contents of a trade
+	pub fn verify(&self) -> bool {
+		// TODO: Verify the signatures of both orders
```
Still pending

Gauthamastro replied:
Fixed

serhii-temchenko commented on primitives/orderbook/src/types.rs:
```diff
@@ -0,0 +1,595 @@
+			main_account: self.main.clone(),
+			amount: self.amount()?,
+			asset: self.payload.asset_id,
+			event_id: 0,              // TODO: We don't use this
+			fees: Default::default(), // TODO: We don't use this
```
Still pending

Gauthamastro replied:
Resolved

serhii-temchenko commented on primitives/orderbook/src/types.rs:
```diff
@@ -0,0 +1,595 @@
+		let mut temp = self.avg_filled_price.saturating_mul(self.filled_quantity);
+		temp = temp.saturating_add(amount.saturating_mul(price));
+		self.filled_quantity = self.filled_quantity.saturating_add(amount);
+		println!("self.filled_quantity: {:?}\ntemp: {:?}", self.filled_quantity, temp);
```
Do we need this print?

Gauthamastro replied:
Prints and unnecessary logs can be removed after testinng, during final clean up before release to main

serhii-temchenko commented on primitives/orderbook/src/types.rs:
```diff
@@ -0,0 +1,595 @@
+		println!("self.filled_quantity: {:?}\ntemp: {:?}", self.filled_quantity, temp);
+		match temp.checked_div(self.filled_quantity) {
+			Some(quotient) => {
+				println!("Quotient: {:?}", quotient);
```
Do we need this print?

Gauthamastro replied:
Same as above

serhii-temchenko commented on primitives/orderbook/src/types.rs:
```diff
@@ -0,0 +1,595 @@
+	pub fn available_volume(&self, other_price: Option<Decimal>) -> Decimal {
+		//this if for market bid order
+		if self.qty.is_zero() {
+			println!(
+				"quote_order_qty: {:?}, avg_filled_price: {:?}, filled_quantity: {:?}",
+				self.quote_order_qty, self.avg_filled_price, self.filled_quantity
+			);
```
Do we need this print? why not used logger?

Gauthamastro replied:
Same as above

serhii-temchenko commented on primitives/orderbook/src/types.rs:
```diff
@@ -0,0 +1,595 @@
+	pub fn test_ob_message() {
+		let msg = ObMessage { stid: 0, action: UserActions::BlockImport(1) };
+
+		println!("OBMessage: {:?}", serde_json::to_string(&msg).unwrap());
```
Do we need this print? why not used logger?

Gauthamastro replied:
Same as above

serhii-temchenko commented on primitives/polkadex/Cargo.toml:
```diff
@@ -0,0 +1,43 @@
+sp-runtime = { default-features = false, git = "https://github.com/paritytech/substrate.git", branch = "polkadot-v0.9.28" }
+sp-application-crypto = { default-features = false, git = "https://github.com/paritytech/substrate.git", branch = "polkadot-v0.9.28" }
+frame-support = { default-features = false, git = "https://github.com/paritytech/substrate.git",  branch = "polkadot-v0.9.28" }
+rust_decimal = {git ="https://github.com/Polkadex-Substrate/rust-decimal.git", branch="master", features = ["scale-codec"], default-features = false }
```
Formatting

serhii-temchenko replied:
Fixed

serhii-temchenko commented on primitives/polkadex/Cargo.toml:
```diff
@@ -0,0 +1,43 @@
+edition = "2021"
+
+[dependencies]
+primitive-types = {version = "0.12.1", default-features =false, features = ["scale-info"]}
```
Formatting

serhii-temchenko replied:
Fixed

serhii-temchenko commented on primitives/polkadex/src/assets.rs:
```diff
@@ -0,0 +1,167 @@
+// This file is part of Polkadex.
+
+// Copyright (C) 2020-2021 Polkadex oü.
```
Please change to 2023

serhii-temchenko replied:
Fixed

serhii-temchenko commented on primitives/polkadex/src/ingress.rs:
```diff
@@ -0,0 +1,77 @@
+pub struct HandleBalanceLimit;
+
+impl Get<u32> for HandleBalanceLimit {
+	//ToDo: Set an arbitrary value to 1000.
```
Still pending

Gauthamastro replied:
We can ignore it, as it is part of v1 arch.

serhii-temchenko commented on primitives/polkadex/src/lib.rs:
```diff
@@ -0,0 +1,131 @@
+// This file is part of Polkadex.
+
+// Copyright (C) 2020-2021 Polkadex oü.
```
Please change to 2023

serhii-temchenko replied:
Fixed

serhii-temchenko commented on primitives/polkadex/src/lib.rs:
```diff
@@ -0,0 +1,131 @@
+/// Block ID.
+pub type BlockId = generic::BlockId<Block>;
+
+// TODO: Figure out the actual bound given below
```
Still pending

Gauthamastro replied:
3 is the final value, removing TODO.

serhii-temchenko commented on prometheus.yml:
```diff
@@ -0,0 +1,8 @@
+    scrape_interval: 5s
+
+    static_configs:
+      - targets: ["localhost:9615"]
```
Please add empty string at the end of file

serhii-temchenko replied:
Fixed

serhii-temchenko commented on runtime/Cargo.toml:
```diff
@@ -96,6 +96,7 @@ pdex-migration = { path = "../pallets/pdex-migration", default-features = false
 asset-handler = { path = "../pallets/asset-handler", default-features = false }
 chainbridge = { path ="../pallets/chainbridge", default-features = false }
 thea-cross-chain-governance = { path = "../pallets/thea-cross-chain-governance", default-features = false }
+orderbook-primitives = {path = "../primitives/orderbook", default-features = false}
```
Formatting

serhii-temchenko replied:
Fixed

serhii-temchenko commented on runtime/Cargo.toml:
```diff
@@ -96,6 +96,7 @@ pdex-migration = { path = "../pallets/pdex-migration", default-features = false
 asset-handler = { path = "../pallets/asset-handler", default-features = false }
 chainbridge = { path ="../pallets/chainbridge", default-features = false }
```
Formatting

serhii-temchenko replied:
Fixed

serhii-temchenko commented on scripts/start_chain.sh:
```diff
@@ -0,0 +1,134 @@
+
+trap kill_nodes EXIT
+
+start_chain
```
Empty line at the end of file

serhii-temchenko replied:
Fixed

serhii-temchenko reviewed (changes requested):
A lot of unresolved todos are present in this PR. I assume this should be merged asap (basically now) and there is no time to address them all, so these comments which points todos should be used as a reference to get back and resolve them after merging this PR. 
Plus really poor documentation of the public API. It should be added as well at some point.

35359595 committed:
few benchmarks cleanups

35359595 committed:
Merge branch 'gj-ob' of github.com:Polkadex-Substrate/Polkadex into gj-ob

35359595 committed:
benchmarks for ocex

Gauthamastro committed:
Merge branch 'gj-ob' into fx-remove-shutdown

Gauthamastro committed:
Remove Shutdown message (#678)

This PR removes the `Shutdown` Ingress Message and extrinsic for the
same until we can figure out how we deal with such messages.

Gauthamastro committed:
Remove unused dependencies

35359595 committed:
added benchmarks for new extrinsics and old clean up

Gauthamastro committed:
tests compile now

35359595 committed:
snapshot generator

Gauthamastro committed:
test_submit_snapshot is fixed

Gauthamastro committed:
test_submit_snapshot_bad_origin is fixed

Gauthamastro committed:
test_submit_snapshot_snapshot_nonce_error is fixed

Gauthamastro committed:
test_withdrawal is fixed

Gauthamastro committed:
collect_fees is fixed

Gauthamastro committed:
test_onchain_events_overflow is fixed

Gauthamastro commented:
The failing OCEX test cases are fixed in PR #694

35359595 committed:
tests build

35359595 committed:
ocex clippi fixes

35359595 committed:
orderbook primitives clippy fixes

35359595 committed:
primitives clippy fixes

35359595 committed:
ocex tests clippy fixes

35359595 committed:
primitives tests clippy fixes

35359595 committed:
orderbook primitives tests clippy fixes

35359595 committed:
crosschain gov tests clippy fixes

35359595 committed:
proper enum values

35359595 committed:
node tests built

35359595 committed:
node clippy fixes

35359595 committed:
orderbook tests clippy fixes

35359595 committed:
fmt applied

35359595 committed:
warnings no more

35359595 committed:
Fix OCEX tests. (#694)

35359595 committed:
Merge branch 'gj-ob' of github.com:Polkadex-Substrate/Polkadex into gj-ob

35359595 committed:
ocex benchmarks running again

35359595 committed:
ocex updated benchmarks and new weights

serhii-temchenko committed:
Code style fixes

serhii-temchenko committed:
Updated copyright documentation

serhii-temchenko committed:
Removed commented out code

serhii-temchenko committed:
Removed temporary added branches from workflow events trigger

serhii-temchenko committed:
Code style fixes

serhii-temchenko committed:
Reverted temporary changes of build in the CI

Gauthamastro committed:
Update fixes

Gauthamastro committed:
Update fixes

Gauthamastro committed:
cargo fmt

Gauthamastro committed:
Fix tests

Gauthamastro committed:
fix benchmarking code

Gauthamastro committed:
remove warnings and cargo fmt
